src/frontends/tensorflow_common/src/op/reverse_sequence.cpp

```diff
@@ -14,19 +15,30 @@ namespace frontend {
 namespace tensorflow {
 namespace op {
 OutputVector translate_reverse_sequence_op(const NodeContext& node) {
-    default_op_checks(node, 2, {"ReverseSequence", "REVERSE_SEQUENCE"});
+    default_op_checks(node, 2, {"ReverseSequence", "REVERSE_SEQUENCE"}, true);
     auto input = node.get_input(0);
     auto seq_lengths = node.get_input(1);
+    auto complex_type_mark = as_type_ptr<ComplexTypeMark>(input.get_node_shared_ptr());
 
     // retrieve attributes
     auto seq_dim = node.get_attribute<int64_t>("seq_dim");
     auto batch_dim = node.get_attribute<int64_t>("batch_dim", 0);
 
+    if (complex_type_mark) {
+        element::Type complex_part_type = complex_type_mark->get_complex_part_type();
+        input = complex_type_mark->input_value(0);
+
+        auto reverse_sequence = make_shared<v0::ReverseSequence>(input, seq_lengths, batch_dim, seq_dim);
```

Comment:
I think we need to adjust `batch_dim` and `seq_dim` if they are negative because aux dim for complex type affects this.

if `batch_dim < 0` then `batch_dim -= 1`. the same for `seq_dim`